From d15fa18c46c5e1f9c5fd021538db4b983d6eb653 Mon Sep 17 00:00:00 2001
From: Evan Benn <evan.benn@morsemicro.com>
Date: Mon, 19 May 2025 15:55:18 +1000
Subject: [PATCH] APP-4789: iwinfo: Use IWINFO_ASSOCLIST_BUFSIZE for buffer

As the buffer is used for assoclist calls use the assoclist buffer
size.
---
 src/iwinfo.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/iwinfo.c b/src/iwinfo.c
index e6c48c24bd90..38310d871d8a 100644
--- a/src/iwinfo.c
+++ b/src/iwinfo.c
@@ -74,7 +74,12 @@ static void iwinfo_submit(const char *ifname, const char *type, int value)
 static void iwinfo_process(const char *ifname)
 {
 	int val;
-	char buf[IWINFO_BUFSIZE];
+	char *buf = malloc(IWINFO_ASSOCLIST_BUFSIZE);
+	if (buf == NULL)
+	{
+		WARNING("iwinfo: malloc failed");
+		return;
+	}
 	const struct iwinfo_ops *iw = iwinfo_backend(ifname);
 
 	/* does appear to be a wifi iface */
@@ -103,6 +108,7 @@ static void iwinfo_process(const char *ifname)
 	}
 
 	iwinfo_finish();
+	free(buf);
 }
 
 static int iwinfo_read(void)
-- 
2.48.1

